doc/1.4/language.md

```diff
@@ -240,33 +240,36 @@ A device is made up of a number of member objects and methods, where any object
 may contain further objects and methods of its own. Many object types only make
 sense in a particular context and are not allowed elsewhere:
 
-* Objects of type [`bank`](#register-banks) or [`connect`](#connects) may only
-  appear as part of a `device`.
+* There is exactly one [`device`](#the-device) object. It resides on the top
+  level.
+
+* Objects of type [`bank`](#register-banks), [`port`](#ports) or
+  [`subdevice`](#subdevices) may only appear as part of a `device` or
+  `subdevice` object.
 
 * Objects of type [`implement`](#implements) may only appear as part of a
-`device`, `port` or `bank`.
+  `device`, `port`, `bank`, or `subdevice` object.
 
-* Objects of type [`group`](#groups) may appear anywhere.
+* Objects of type [`register`](#registers) may only appear as part of a `bank`.
 
-* Objects of type [`register`](#registers) may only appear as part of
-a `bank`, or indirectly as part of a (possibly nested) `group` that
-is part of a `bank`.
+* Objects of type [`field`](#fields) may only appear as part of a `register`.
 
-* Objects of type [`field`](#fields) may only appear as part of a `register`, or
-indirectly as part of a (possibly nested) `group` that is part of a
-`register`.
+* Objects of type [`connect`](#connects) may only appear as part of a `device`,
+  `subdevice`, `bank`, or `port` object.
 
-* Objects of type [`interface`](#interfaces) may only appear as part of a
-  `connect`.
+* Objects of type [`interface`](#interfaces) may only appear directly below a
+  `connect` object.
 
 * Objects of type [`attribute`](#attributes) may only appear as part of a
-  `device`, `bank`, `port`, or `implement`, or indirectly as part of a (possibly
-  nested) `group` that is part of an object of any one of these types.
+  `device`, `bank`, `port`, or `implement` object.
 
 * Objects of type [`event`](#events) may appear anywhere **except** as part of a
-  `field`, `interface`, `implement`, or another `event`. They may also not
-  appear indirectly as part of a (possibly nested) `group` that is part of an
-  object of any one of these types.
+  `field`, `interface`, `implement`, or another `event`.
+
+* Objects of type [`group`](#groups) are neutral: Any object may contain a
+  `group` object, and a `group` object may contain any object that its parent
+  object may contain, with the exception that a `group` cannot contain an
+  object of type `interface` or `implement`.
```

Comment:
Add an item about `subdevice`!

Reply:
See lines 247-248.

Reply:
What this list is missing is an item telling you what `subdevice` may appear in. 247-248 just tells you an implement may appear as part of a `subdevice`.

Also, I just noticed that `event`, `attribute`, `connect` doesn't say they may appear within a `subdevice`.